### PR TITLE
(CDAP-2293) Change template and plugin directory naming

### PR DESCRIPTION
**Pull request by chtyim**
- Also enable the StreamConversionTest by fixing the export-package of the sink plugin.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/resources/etl.properties [hunk 1]
```diff
@@ -13,5 +13,5 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 #
-etl.batch.plugin.name=etlBatch
-etl.realtime.plugin.name=etlRealtime
+etl.batch.plugin.name=${etl.batch.plugin.name}
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/resources/etl.properties]: i think the batch and realtime templates read this file to set their names. Does this get properly expanded for them?

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/BaseETLBatchTest.java [hunk 2]
```diff
@@ -52,7 +54,7 @@ public static void setupTest() throws IOException {
       DBSource.class, KVTableSource.class, StreamBatchSource.class, TableSource.class);
     addTemplatePlugins(TEMPLATE_ID, "batch-sinks-1.0.0.jar",
       BatchCubeSink.class, DBSink.class, KVTableSink.class,
-      TableSink.class, TimePartitionedFileSetDatasetAvroSink.class);
+      TableSink.class, TimePartitionedFileSetDatasetAvroSink.class, AvroKeyOutputFormat.class, AvroKey.class);
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/BaseETLBatchTest.java]: do these packages need to be added as exports to the pom?

File: cdap-app-templates/cdap-etl/cdap-etl-core/src/main/resources/etl.properties (hunk 1 above)
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-core/src/main/resources/etl.properties]: The value is set in maven pom property. The maven-resources-plugin will do the substitution. By doing this, we have one place to define the names.

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/BaseETLBatchTest.java (hunk 2 above)
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/BaseETLBatchTest.java]: Yes, they were added in a separate PR https://github.com/caskdata/cdap/pull/2480

**Comment by albertshau** [comment]
:+1:

**Comment by sreevatsanraman** [comment]
Build passed. Merging. http://builds.cask.co/browse/CDAP-RBT263

**Pull request merged**